docs/src/book.md

```diff
@@ -0,0 +1,196 @@
+
+Generic character tables were so far exclusive to the CHEVIE system
+(see e.g. [MR1486215](@cite) for details),
+which involves use of an old version of \Maple
+which heavily restricts the access to this data.
+Attempts to port this
+to e.g. \GAP in the past were never published due to it being considered too slow.
+
+Thanks to Julia's impressive performance,
+we were able to overcome this obstacle with data imported from CHEVIE.
+The code is still in early stages
+as we write this, but this situation is rapidly improving -- so be sure to
+check <https://book.oscar-system.org> for updated examples for this section.
+
+Right now the code is in a separate repository. To use it, enter within a Julia interactive command-line (REPL):
+```
+julia> import Pkg; Pkg.add("GenericCharacterTables"); using GenericCharacterTables
+```
+
+In the following we replicate parts of Section 5.3 of [MR1486215](@cite),
+and strongly recommend to read that concurrently with the present text.
+
+To start we load the generic character table for $\mathrm{SL}_3(q)$ with
+$q\not\equiv 1\pmod 3$.
+To learn about the origin of the table, we could enter `printinfotab(T)`.
+```jldoctest book
+julia> T = genchartab("SL3.n1")
+Generic character table SL3.n1
+  of order q^8 - q^6 - q^5 + q^3
+  with 8 irreducible character types
+  with 8 class types
+  with parameters (a, b, m, n)
+```
+
+As we can see this table has four parameters in addition to $q$.
+The entries of the table are “generalized cyclotomics”, that is,
+linear combinations over $\QQ(q)$ of symbolic “roots of unity”
+depending on the parameters listed above.
+```jldoctest book
+julia> printval(T,char=4,class=4)
+Value of character type 4 on class type
+  4: (q + 1)*exp(2π𝑖(1//(q - 1)*a*n)) + exp(2π𝑖(-2//(q - 1)*a*n))
+```
+
+Denoting row 4 of the table by $\chi_4$, we note that is not a single
+character, but a *character type*, describing a whole family of
+characters.
+We can compute their norms, scalar products, and more. For this demonstration
+we tensor the second character type $\chi_2$ with itself.
+```jldoctest book
+julia> h = tensor(T[2],T[2])
+Generic character of SL3.n1
+  of degree q^4 + 2*q^3 + q^2
+  with values
+    q^4 + 2*q^3 + q^2
+    q^2
+    0
+    q^2 + 2*q + 1
+    1
+    4
+    0
+    1
+```
+
+We may now attempt to decompose this character type `h` by computing its scalar product with the irreducible character types.
+This returns a “generic” scalar product, plus a (possibly empty)
+list of parameter exceptions for which the general result may not hold.
+For example:
+```jldoctest book
+julia> scalar(T[4], h)
+0
+With exceptions:
+  2*n1 ∈ (q - 1)ℤ
+```
+This scalar product is $0$
+except when $q-1$ divides $2n_1$, where $n_1$ indicates the value of the parameter $n$ for the first factor in the scalar product (i.e. the character type $\chi_4$).
+For a “generic decomposition” we need to compute all the scalar products and
+exceptions.
+```jldoctest book
+julia> print_decomposition(T, h)
```

Comment:
Thanks to the new printing I might also just replace this by, say,
```suggestion
julia> for i in 1:8 println(i, ": ", scalar(T[i], h)) end
```
which of course produces a bit different output but it's really good enough

Reply:
Can't we remove `print_decomposition` then?

Reply:
done